onnxruntime/core/providers/cuda/cuda_common.h

```diff
@@ -63,6 +63,9 @@ class CudaKernel : public OpKernel {
   }
 
   Status Compute(OpKernelContext* p_op_kernel_context) const override {
+    // use mutex to make sure memory allocation/usage within a kernel is not mixed with another kernel
+    std::lock_guard<std::mutex> lock(provider_->GetKernelMutex());
+
     auto s = ComputeInternal(p_op_kernel_context);
```

Comment:
What's the performance impact of taking a lock across all the whole Compute call if you have concurrent requests? I would have expected that to reduce throughput given they're all using the same provider instance.

Reply:
Reverted. The mutex is not needed.